### PR TITLE
feat(sage-monorepo): add GH workflow to link Dockerfile

### LINKED ISSUE #2572: [Story] Add GH workflow to link Dockerfile
### What product(s) is this story for?

Sage Monorepo

### As a user, I want

_No response_

### Description

_No response_

### Acceptance criteria

_No response_

### Tasks

_No response_

### Anything else?

_No response_

### Have you linked this story to a GitHub Project?

- [X] I have linked this story to a GitHub Project and set its metadata.

### PR DESCRIPTION
**Pull request by tschaffter**
Closes #2572

## Changelog

- add Hadolint config file `.hadolint.yaml`
- fix issues identified by Hadolint in all Dockerfiles
- add GitHub workflow to lint Dockerfiles

### CODE REVIEW

**Comment by tschaffter** [comment]
## Hadolint extension

The hadolint extension for VS Code requires hadolint to be installed.

> hadolint: Cannot find hadolint from system $PATH. Please install hadolint.

**Review by vpchung — APPROVED** [review]
LGTM.

**Comment by linglp** [comment]
@tschaffter is the expectation that we will name branches related to schematic following patterns: `schematic/**`? FAIR data team currently likes to use Jira ticket number or feature name to name a branch.. but if the expectation is to name branches with pre-fix `schematic/**`, we could adjust. And as dev, I kind of like to be able to trigger a GH action manually for a specific branch. Could you add that too?

File: apps/schematic/api/Dockerfile [hunk 1]
```diff
@@ -13,21 +13,21 @@ ENV PYTHONFAULTHANDLER=1 \
...
     VERSION=$TAG
 
 # run open ssl and generate certificate
-RUN apt update \
+RUN apt-get update -qq -y && export DEBIAN_FRONTEND=noninteractive \
+    && apt-get install --no-install-recommends -qq -y \
+      openssl jq \
     && apt-get -y autoclean \
     && apt-get -y autoremove \
     && rm -rf /var/lib/apt/lists/* \
```
Comment (linglp) [inline apps/schematic/api/Dockerfile:30]: shouldn't that `\` be removed? there's nothing after `\`

**Review by linglp — CHANGES_REQUESTED** [review]

File: apps/schematic/api/Dockerfile (hunk 1 above)
Comment (tschaffter) [reply apps/schematic/api/Dockerfile:30]: You're right. Removing it now...

Comment (tschaffter) [reply apps/schematic/api/Dockerfile]: Done

**Review by linglp — APPROVED** [review]

**Pull request merged**